### DISCUSSION TITLE
Crazyflie 2.1 with bigquad deck shut down immediately after take off

### DISCUSSION THREAD

**Discussion by wdliu356** [bitcraze/discussions#1604] [Q&A]
Hi all
I'm working on some customized firmware with crazyflie 2.1 and bigquad deck. I've verified my firmware with crazyflie 2.1 only already and it is working well. However, when I added bigquad deck, it worked well with low thrust but it shut down automatically when I send a higher thrust command or maybe more aggressive attitude command. Is there any automatic shut-down command built in for the bigquad deck that I didn't realize?

**Comment by wdliu356** [comment]
I used a new crazyflie 2.1 board and now it cannot run from the very beginning, which means it cannot react to my command at all. So I double checked the source code of the firmware and found out there's a supervisor used in stabilizer.c. I believe that's the reason why it didn't react to my setpoint command. Can you explain in brief how is that working and what might possibly lead to the failure of passing the supervisor.

**Reply by gemenerik** [reply]
Can you elaborate a bit more? What supervisor state are you in? Crashed? Tumbled? Locked? Here is the [documentation for the supervisor](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/supervisor/).

**Reply by wdliu356** [reply]
Sorry for a late reply. I switched back to my original board and I found the supervisor state is tumbled when it shut down. So i checked the code of tumbled state and was a bit confused about how it checks whether the drone is tilted. Why it is using `sensor.acc.z` to check that? Why not reading from gyro to check that?

**Reply by wdliu356** [reply]
I did experiment today again and I found sometimes it shut down automatically when `supervisor.info = 40` which is in fly mode. Now I got no clue at all for why it is shutting down itself....

**Reply by wdliu356** [reply]
I commented out all the safety check in `stabilizer.c` but it still shuts down itself sometimes. Is there any other safety check in the code that I didn't notice?

**Reply by wdliu356** [reply]
I believe the problem is caused by the communication issue between the quad and my computer. It sometimes failed to set parameter through crazyflie-lib-python. I got this conclusion by having it print the parameters when the `param.set_value` function worked and sometimes it printed nothing and the drone didn't respond to my command at all. However, the communication is connected at that time since I could get reading from `supervisor.info` and the value is always 40, indicating the supervisor think the drone was flying. I can get the supervisor reading even before I can successfully set my parameter, which is super weird. Do you have any idea for that?

**Reply by wdliu356** [reply]
After some more tests, I found that the drone failed to receive my command after flying in the air for some time. All the command goes to 0 and onboard command was set to default value automatically, which led to the crash of my drone. Do you have any idea about what could possibly cause this problem? This didn't happen when I was not using the big quad deck.

**Reply by tobbeanton** [reply]
This is all hard to tell but I'm thinking it could be an electrical problem or electrical noise. Could also be a link issue. Did you loose the link when it happened?

**Reply by wdliu356** [reply]
I think the link is still on since I can still read the roll, pitch and yaw log from the drone.

**Reply by wdliu356** [reply]
What I did is reading the command from the log in the controller. Is there any prior step that would adjust the setpoint before entering the controller? I know the supervisor can adjust the setpoint but what I got from `supervisor.info` was always 40 indicating supervisor shouldn't have adjusted the setpoint. Are there places other than that might adjust setpoint?

**Reply by tobbeanton** [reply]
It is a bit hard to tell what the problem is but my bet is that it isn't the supervisor, especially if you turned off the tumble detection.

**Reply by wdliu356** [reply]
I turned that back on and supervisor.info continuously give me 40 which is flying

**Reply by tobbeanton** [reply]
I have two hypothothesis. 
- The big-quad has a lot of vibrations that couse the supervisor to bahave strangly.
- The big-quad has power supply problems cousing the crazyflie to reset/reboot or drop the link.

if you can not replicate the problem when running on a bara crazyflie it points even more to the above?

**Reply by wdliu356** [reply]
I thought about the power supply issue before so I added an additional battery that directly powered the flight controller and it didn't solve the problem. So probably not a power issue. Might be a vibration issue but I didn't get any negative feedback from the supervisor. Are you advising me to comment that out for a test? I believe I've done that before but still not doing any help.

**Reply by tobbeanton** [reply]
If the supervisor triggers you should see this and it will also go into the locked state when it triggers. Are you compiling the firmware with auto-arming or arming-required? So as you say, the supervisor is most likely not the problem. As you say it also works on a normal crazyflie (without big-quad) I'm even more leaning towards a electrical problem. Could e.g. be that something glitches.

One possible test. Can you fly with it manually and see if you trigger the same problem?

**Reply by wdliu356** [reply]
Yeah that makes sense. I'll keep testing that in the following weeks.
